### ISSUE TITLE
`unused-function-argument` - false negative when argument is reassigned before use

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ruff#27537]
### Summary

```py
def foo(value: int): # no error
    value = 1
    print(value)
```

(can't provide a playground link due to #22834)

### Version

0.16.1

**Comment by Hawkvs** [comment]
@DetachHead I tested the reproducer from the issue on the current main branch and couldn't reproduce the reported behavior. Running ruff check --select ARG001 on the provided example reports "All checks passed". Is this issue specific to v0.16.1, or is there another reproducer that still triggers the false positive?

**Comment by DetachHead** [comment]
oops i meant false negative, updated the title

**Comment by Hawkvs** [comment]
@DetachHead  Thanks for the clarification! I understand the expected behavior now. I'd like to investigate this issue by looking into how parameter bindings and subsequent assignments are tracked in Ruff's semantic analysis.
